caddyhttp/staticfiles/fileserver.go

```diff
@@ -107,7 +107,7 @@ func (fs FileServer) serveFile(w http.ResponseWriter, r *http.Request) (int, err
 		// if an index file was explicitly requested, strip file name from the request
 		// ("/foo/index.html" -> "/foo/")
 		for _, indexPage := range IndexPages {
-			if strings.HasSuffix(urlCopy.Path, indexPage) {
+			if path.Base(urlCopy.Path) == indexPage {
```

Comment:
For a very, very, very minor optimization, `path.Base(urlCopy.Path)` can be computed just before the `for` loop.